From ac245dccc0edcbd94fcd465e77bef3f9cd05f622 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Tue, 6 Aug 2019 15:46:10 +0800
Subject: [PATCH] Add WRLinux1019 specific oval check for rule
 file_owner_cron_allow

Add WRLinux1019 specific oval check for rule file_owner_cron_allow

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../file_owner_cron_allow/oval/wrlinux1019.xml     | 31 ++++++++++++++++++++++
 1 file changed, 31 insertions(+)
 create mode 100644 linux_os/guide/services/cron_and_at/restrict_at_cron_users/file_owner_cron_allow/oval/wrlinux1019.xml

diff --git a/linux_os/guide/services/cron_and_at/restrict_at_cron_users/file_owner_cron_allow/oval/wrlinux1019.xml b/linux_os/guide/services/cron_and_at/restrict_at_cron_users/file_owner_cron_allow/oval/wrlinux1019.xml
new file mode 100644
index 0000000..6fd7218
--- /dev/null
+++ b/linux_os/guide/services/cron_and_at/restrict_at_cron_users/file_owner_cron_allow/oval/wrlinux1019.xml
@@ -0,0 +1,31 @@
+<def-group>
+  <definition class="compliance"
+  id="file_owner_cron_allow" version="1">
+    <metadata>
+      <title>Verify user who owns 'cron.allow' file</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The /etc/cron.allow file should be owned by the
+      appropriate user.</description>
+    </metadata>
+    <criteria>
+      <criterion comment="Check file ownership of /etc/cron.allow"
+      test_ref="test_userowner_cron_allow_file" />
+    </criteria>
+  </definition>
+  <unix:file_test check="all" check_existence="any_exist"
+  comment="Testing user ownership of /etc/cron.allow"
+  id="test_userowner_cron_allow_file" version="1">
+    <unix:object object_ref="object_file_etc_cron_allow" />
+    <unix:state state_ref="state_etc_cron_allow_uid_root" />
+  </unix:file_test>
+  <unix:file_state id="state_etc_cron_allow_uid_root"
+  version="1">
+    <unix:user_id datatype="int">0</unix:user_id>
+  </unix:file_state>
+  <unix:file_object comment="/etc/cron.allow"
+  id="object_file_etc_cron_allow" version="1">
+    <unix:filepath>/etc/cron.allow</unix:filepath>
+  </unix:file_object>
+</def-group>
-- 
1.9.1

